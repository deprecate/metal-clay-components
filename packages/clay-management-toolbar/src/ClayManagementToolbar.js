import 'clay-button';
import 'clay-checkbox';
import 'clay-dropdown';
import Component from 'metal-component';
import defineWebComponent from 'metal-web-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import itemsValidator from './items_validator';
import templates from './ClayManagementToolbar.soy.js';

/**
 * Metal ClayManagementToolbar component.
 */
class ClayManagementToolbar extends Component {
	/**
	 * Continues the propagation of the checkbox changed event
	 * @param {!Event} event
	 * @private
	 */
	handleFilterDoneButtonClick_(event) {
		this.emit('filterDone', event);
	}

	/**
	 * Continues the propagation of the plus button clicked event
	 * @param {!Event} event
	 * @private
	 */
	handlePlusButtonClicked_(event) {
		this.emit('plusButtonClicked', event);
	}

	/**
	 * Continues the propagation of the select all button clicked event
	 * @param {!Event} event
	 * @private
	 */
	handleSelectAllClicked_(event) {
		this.emit('selectAllClicked', event);
	}

	/**
	 * Continues the propagation of the checkbox changed event
	 * @param {!Event} event
	 * @private
	 */
	handleSelectPageCheckboxChanged_(event) {
		this.emit('selectPageCheckboxChanged', event);
	}

	/**
	 * Continues the propagation of the sorting button clicked event
	 * @private
	 */
	handleSortingButtonClicked_() {
		this.emit('sortingButtonClicked', {
			sortingOrder: this.sortingOrder,
		});
	}
}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayManagementToolbar.STATE = {
	/**
	 * List of items to display in the actions menu on active state.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?array|undefined}
	 * @default undefined
	 */
	actionItems: itemsValidator,

	/**
	 * CSS classes to be applied to the element.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?string|undefined}
	 * @default undefined
	 */
	elementClasses: Config.string(),

	/**
	 * List of filter menu items.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?array|undefined}
	 * @default undefined
	 */
	filterItems: itemsValidator,

	/**
	 * Id to be applied to the element.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?string|undefined}
	 * @default undefined
	 */
	id: Config.string(),

	/**
	 * Flag to indicate if the managment toolbar will control the selection of
	 * elements.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?bool|undefined}
	 * @default undefined
	 */
	selectable: Config.bool().value(false),

	/**
	 * Number of selected items.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?number|undefined}
	 * @default undefined
	 */
	selectedItems: Config.number(),

	/**
	 * Sorting order.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?string|undefined}
	 * @default asc
	 */
	sortingOrder: Config.oneOf(['asc', 'desc']).value('asc'),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?string|undefined}
	 * @default undefined
	 */
	spritemap: Config.string().required(),

	/**
	 * Total number of items.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?number|undefined}
	 * @default undefined
	 */
	totalItems: Config.number(),

	/**
	 * List of view items.
	 * @instance
	 * @memberof ClayManagementToolbar
	 * @type {?array|undefined}
	 * @default undefined
	 */
	viewTypes: itemsValidator,
};

defineWebComponent('clay-management-toolbar', ClayManagementToolbar);

Soy.register(ClayManagementToolbar, templates);

export {ClayManagementToolbar};
export default ClayManagementToolbar;
