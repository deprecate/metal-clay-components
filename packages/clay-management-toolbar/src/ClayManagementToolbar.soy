{namespace ClayManagementToolbar}

/**
 * This renders the component's whole content.
 */
{template .render}
	{@param spritemap: string}
	{@param? actionItems: list<?>}
	{@param? elementClasses: string}
	{@param? filterItems: list<?>}
	{@param? handleFilterDoneButtonClick_: any}
	{@param? handlePlusButtonClicked_: any}
	{@param? handleSelectAllClicked_: any}
	{@param? handleSelectPageCheckboxChanged_: any}
	{@param? handleSortingButtonClicked_: any}
	{@param? id: string}
	{@param? selectable: bool}
	{@param? selectedItems: number}
	{@param? sortingOrder: string}
	{@param? totalItems: number}
	{@param? viewTypes: list<?>}

	{let $isActive: $selectedItems and $selectedItems > 0 /}

	{if $isActive}
		{call .active}
			{param actionItems: $actionItems /}
			{param elementClasses: $elementClasses /}
			{param handleSelectAllClicked_: $handleSelectAllClicked_ /}
			{param handleSelectPageCheckboxChanged_: $handleSelectPageCheckboxChanged_ /}
			{param selectedItems: $selectedItems /}
			{param spritemap: $spritemap /}
			{param totalItems: $totalItems /}
		{/call}
	{else}
		{call .default}
			{param elementClasses: $elementClasses /}
			{param filterItems: $filterItems /}
			{param handleFilterDoneButtonClick_: $handleFilterDoneButtonClick_ /}
			{param handlePlusButtonClicked_: $handlePlusButtonClicked_ /}
			{param handleSelectPageCheckboxChanged_: $handleSelectPageCheckboxChanged_ /}
			{param handleSortingButtonClicked_: $handleSortingButtonClicked_ /}
			{param id: $id /}
			{param selectable: $selectable /}
			{param sortingOrder: $sortingOrder /}
			{param spritemap: $spritemap /}
			{param viewTypes: $viewTypes /}
		{/call}
	{/if}
{/template}

/**
 * This renders the component's active content.
 */
{template .active}
	{@param spritemap: string}
	{@param? actionItems: list<?>}
	{@param? elementClasses: string}
	{@param? handleSelectAllClicked_: any}
	{@param? handleSelectPageCheckboxChanged_: any}
	{@param? id: string}
	{@param? selectedItems: number}
	{@param? totalItems: number}

	{let $navAttributes kind="attributes"}
		class="management-bar management-bar-primary navbar navbar-expand-md
			{if $elementClasses}
				{sp}{$elementClasses}
			{/if}
		"

		{if $id}
			id="{$id}"
		{/if}
	{/let}

	<nav {$navAttributes}>
		<div class="container">
			<ul class="navbar-nav">
				<li class="nav-item">
					{call ClayCheckbox.render}
						{param checked: true /}
						{param events: ['change': $handleSelectPageCheckboxChanged_] /}
						{param hideLabel: true /}
						{param indeterminate: $selectedItems < $totalItems /}
						{param label: 'Select Page' /}
						{param ref: 'selectPageCheckbox' /}
					{/call}
				</li>
				<li class="nav-item">
					<span class="navbar-text">
						{$selectedItems} of {$totalItems}{sp}
						<span class="navbar-breakpoint-down-d-none">items selected</span>
					</span>
				</li>
				<li class="nav-item">
					{call ClayButton.render}
						{param elementClasses: 'nav-btn' /}
						{param events: ['click': $handleSelectAllClicked_] /}
						{param label: 'Select all' /}
						{param ref: 'selectAllButton' /}
						{param style: 'link' /}
					{/call}
				</li>
			</ul>
			
			{if $actionItems}
				<ul class="navbar-nav">
					<li class="nav-item">
						{call ClayActionsDropdown.render}
							{param items: $actionItems /}
							{param spritemap: $spritemap /}
						{/call}
					</li>
				</ul>
			{/if}
		</div>
	</nav>
{/template}

/**
 * This renders the component's active content.
 */
{template .default}
	{@param spritemap: string}
	{@param? elementClasses: string}
	{@param? filterItems: list<?>}
	{@param? handleFilterDoneButtonClick_: any}
	{@param? handlePlusButtonClicked_: any}
	{@param? handleSelectPageCheckboxChanged_: any}
	{@param? handleSortingButtonClicked_: any}
	{@param? id: string}
	{@param? selectable: bool}
	{@param? sortingOrder: string}
	{@param? viewTypes: list<?>}

	{let $navAttributes kind="attributes"}
		class="management-bar management-bar-light navbar navbar-expand-md
			{if $elementClasses}
				{sp}{$elementClasses}
			{/if}
		"

		{if $id}
			id="{$id}"
		{/if}
	{/let}

	<nav {$navAttributes}>
		<div class="container">
			<ul class="navbar-nav">
				{if $selectable}
					<li class="nav-item">
						{call ClayCheckbox.render}
							{param events: ['change': $handleSelectPageCheckboxChanged_] /}
							{param hideLabel: true /}
							{param label: 'Select Page' /}
							{param ref: 'selectPageCheckbox' /}
						{/call}
					</li>
				{/if}

				{if $filterItems}
					<li class="dropdown nav-item">
						{call ClayDropdown.render}
							{param button: [
								'label': 'Done'
							] /}
							{param events: ['buttonClicked': $handleFilterDoneButtonClick_] /}
							{param items: $filterItems /}
							{param label: 'Filter and Order' /}
							{param ref: 'filters' /}
							{param spritemap: $spritemap /}
							{param triggerClasses: 'nav-link navbar-breakpoint-down-d-none' /}
							{param type: 'form' /}
						{/call}
					</li>
				{/if}

				<li class="nav-item">
					{let $orderClasses kind="text"}
						nav-link nav-link-monospaced
						{if $sortingOrder == 'desc'}
							{sp}order-arrow-down-active
						{else}
							{sp}order-arrow-up-active
						{/if}
					{/let}

					{call ClayButton.render}
						{param elementClasses: $orderClasses /}
						{param events: ['click': $handleSortingButtonClicked_] /}
						{param icon: [
							'alignment': 'left',
							'spritemap': $spritemap,
							'symbol': 'order-arrow'
						] /}
						{param ref: 'sortingButton' /}
						{param style: 'unstyled' /}
					{/call}
				</li>
			</ul>

			<ul class="navbar-nav">
				{if $viewTypes}
					<li class="dropdown nav-item">
						{call ClayDropdown.render}
							{param icon kind="text"}
								{foreach $viewType in $viewTypes}
									{if $viewType.active}
										{$viewType.icon}
									{/if}
								{/foreach}
							{/param}
							{param items: $viewTypes /}
							{param itemsIconAlignment: 'left' /}
							{param spritemap: $spritemap /}
							{param triggerClasses: 'nav-link nav-link-monospaced' /}
						{/call}
					</li>
				{/if}

				<li class="nav-item">
					{call ClayButton.render}
						{param elementClasses: 'nav-btn nav-btn-monospaced' /}
						{param events: ['click': $handlePlusButtonClicked_ ] /}
						{param icon: [
							'alignment': 'left',
							'spritemap': $spritemap,
							'symbol': 'plus'
						] /}
						{param ref: 'plusButton' /}
					{/call}
				</li>
			</ul>
		</div>
	</nav>
{/template}