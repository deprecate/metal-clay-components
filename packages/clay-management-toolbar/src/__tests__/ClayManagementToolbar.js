import ClayManagementToolbar from '../ClayManagementToolbar';

let managementToolbar;

const spritemap = '../node_modules/lexicon-ux/build/images/icons/icons.svg';

describe('ClayManagementToolbar', function() {
	afterEach(() => {
		if (managementToolbar) {
			managementToolbar.dispose();
		}
	});

	it('should render the default markup', () => {
		managementToolbar = new ClayManagementToolbar({
			spritemap: spritemap,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with classes', () => {
		managementToolbar = new ClayManagementToolbar({
			elementClasses: 'my-custom-class',
			spritemap: spritemap,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with id', () => {
		managementToolbar = new ClayManagementToolbar({
			id: 'myId',
			spritemap: spritemap,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with selection enabled', () => {
		managementToolbar = new ClayManagementToolbar({
			selectable: true,
			spritemap: spritemap,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with ascending sorting', () => {
		managementToolbar = new ClayManagementToolbar({
			sortingOrder: 'asc',
			spritemap: spritemap,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with descending sorting', () => {
		managementToolbar = new ClayManagementToolbar({
			sortingOrder: 'desc',
			spritemap: spritemap,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with filters dropdown', () => {
		managementToolbar = new ClayManagementToolbar({
			filterItems: [
				{
					label: 'Item 1',
					inputName: 'item1checkbox',
					inputValue: '1',
					type: 'checkbox',
				},
				{
					label: 'Item 2',
					inputName: 'item2checkbox',
					inputValue: '2',
					type: 'checkbox',
				},
			],
			spritemap: spritemap,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with view types', () => {
		managementToolbar = new ClayManagementToolbar({
			spritemap: spritemap,
			viewTypes: [
				{
					active: true,
					icon: 'list',
					label: 'List',
				},
				{
					disabled: true,
					icon: 'table',
					label: 'Table',
				},
				{
					icon: 'cards2',
					label: 'Card',
				},
			],
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with state active', () => {
		managementToolbar = new ClayManagementToolbar({
			selectable: true,
			selectedItems: 1,
			spritemap: spritemap,
			totalItems: 10,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with classes in active state', () => {
		managementToolbar = new ClayManagementToolbar({
			elementClasses: 'my-custom-class',
			selectable: true,
			selectedItems: 1,
			spritemap: spritemap,
			totalItems: 10,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with id in active state', () => {
		managementToolbar = new ClayManagementToolbar({
			id: 'myId',
			selectable: true,
			selectedItems: 1,
			spritemap: spritemap,
			totalItems: 10,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a management toolbar with actions in active state', () => {
		managementToolbar = new ClayManagementToolbar({
			actionItems: [
				{
					label: 'Edit',
					href: '#editurl',
				},
				{
					label: 'Delete',
					href: '#deleteurl',
				},
			],
			selectable: true,
			selectedItems: 1,
			spritemap: spritemap,
			totalItems: 10,
		});

		expect(managementToolbar).toMatchSnapshot();
	});

	it('should render a selectable management toolbar and emit an event on checkbox click', () => {
		managementToolbar = new ClayManagementToolbar({
			selectable: true,
			spritemap: spritemap,
		});

		const spy = jest.spyOn(managementToolbar, 'emit');

		managementToolbar.refs.selectPageCheckbox.element
			.querySelector('input')
			.click();

		expect(spy).toHaveBeenCalled();
		expect(spy).toHaveBeenCalledWith(
			'selectPageCheckboxChanged',
			expect.any(Object)
		);
	});

	it('should render a selectable management toolbar and emit an event on checkbox click on active state', () => {
		managementToolbar = new ClayManagementToolbar({
			selectable: true,
			selectedItems: 1,
			spritemap: spritemap,
			totalItems: 10,
		});

		const spy = jest.spyOn(managementToolbar, 'emit');

		managementToolbar.refs.selectPageCheckbox.element
			.querySelector('input')
			.click();

		expect(spy).toHaveBeenCalled();
		expect(spy).toHaveBeenCalledWith(
			'selectPageCheckboxChanged',
			expect.any(Object)
		);
	});

	it('should render a management toolbar and emit an event on sorting button click', () => {
		managementToolbar = new ClayManagementToolbar({
			spritemap: spritemap,
		});

		const spy = jest.spyOn(managementToolbar, 'emit');

		managementToolbar.refs.sortingButton.element.click();

		expect(spy).toHaveBeenCalled();
		expect(spy).toHaveBeenCalledWith(
			'sortingButtonClicked',
			expect.any(Object)
		);
	});

	it('should render a management toolbar and emit an event on plus button click', () => {
		managementToolbar = new ClayManagementToolbar({
			spritemap: spritemap,
		});

		const spy = jest.spyOn(managementToolbar, 'emit');

		managementToolbar.refs.plusButton.element.click();

		expect(spy).toHaveBeenCalled();
		expect(spy).toHaveBeenCalledWith(
			'plusButtonClicked',
			expect.any(Object)
		);
	});

	it('should render a management toolbar and emit an event on select all button click', () => {
		managementToolbar = new ClayManagementToolbar({
			selectable: true,
			selectedItems: 1,
			spritemap: spritemap,
			totalItems: 10,
		});

		const spy = jest.spyOn(managementToolbar, 'emit');

		managementToolbar.refs.selectAllButton.element.click();

		expect(spy).toHaveBeenCalled();
		expect(spy).toHaveBeenCalledWith(
			'selectAllClicked',
			expect.any(Object)
		);
	});

	it('should render a management toolbar and emit an event on filter done button click', () => {
		managementToolbar = new ClayManagementToolbar({
			filterItems: [
				{
					label: 'Item 1',
					inputName: 'item1checkbox',
					inputValue: '1',
					type: 'checkbox',
				},
				{
					label: 'Item 2',
					inputName: 'item2checkbox',
					inputValue: '2',
					type: 'checkbox',
				},
			],
			spritemap: spritemap,
		});

		const spy = jest.spyOn(managementToolbar, 'emit');

		let filters = managementToolbar.refs.filters;
		filters.refs.dropdown.refs.dropdownButton.element.click();

		expect(spy).toHaveBeenCalled();
		expect(spy).toHaveBeenCalledWith(
			'filterDoneClicked',
			expect.any(Object)
		);
	});
});
