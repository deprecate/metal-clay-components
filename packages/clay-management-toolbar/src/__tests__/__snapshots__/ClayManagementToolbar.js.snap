// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayManagementToolbar should render a management toolbar with actions in active state 1`] = `
<nav class="management-bar management-bar-primary navbar navbar-expand-md">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <div class="custom-control custom-checkbox">
          <label>
            <input checked="" class="custom-control-input" ref="input" type="checkbox">
            <span class="custom-control-indicator"></span>
            <span class="custom-control-description sr-only">Select Page</span>
          </label>
        </div>
      </li>
      <li class="nav-item">
        <span class="navbar-text">1 of 10
          <span class="navbar-breakpoint-down-d-none">items selected</span>
        </span>
      </li>
      <li class="nav-item">
        <button class="btn nav-btn btn-link">Select all</button>
      </li>
    </ul>
    <ul class="navbar-nav">
      <li class="nav-item">
        <div class="dropdown dropdown-action">
          <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v">
              <title>ellipsis-v</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#ellipsis-v"></use>
            </svg>
          </button>
          <div class="dropdown-menu">
            <ul class="list-unstyled">
              <li>
                <a class=" dropdown-item" href="#editurl">Edit</a>
              </li>
              <li>
                <a class=" dropdown-item" href="#deleteurl">Delete</a>
              </li>
            </ul>
          </div>
        </div>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with ascending sorting 1`] = `
<nav class="management-bar management-bar-light navbar navbar-expand-md">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-link nav-link-monospaced order-arrow-up-active btn-unstyled" aria-label="order-arrow">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-order-arrow">
            <title>order-arrow</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#order-arrow"></use>
          </svg>
        </button>
      </li>
    </ul>
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-btn nav-btn-monospaced btn-primary" aria-label="plus">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-plus">
            <title>plus</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#plus"></use>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with classes 1`] = `
<nav class="management-bar management-bar-light navbar navbar-expand-md my-custom-class">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-link nav-link-monospaced order-arrow-up-active btn-unstyled" aria-label="order-arrow">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-order-arrow">
            <title>order-arrow</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#order-arrow"></use>
          </svg>
        </button>
      </li>
    </ul>
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-btn nav-btn-monospaced btn-primary" aria-label="plus">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-plus">
            <title>plus</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#plus"></use>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with classes in active state 1`] = `
<nav class="management-bar management-bar-primary navbar navbar-expand-md my-custom-class">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <div class="custom-control custom-checkbox">
          <label>
            <input checked="" class="custom-control-input" ref="input" type="checkbox">
            <span class="custom-control-indicator"></span>
            <span class="custom-control-description sr-only">Select Page</span>
          </label>
        </div>
      </li>
      <li class="nav-item">
        <span class="navbar-text">1 of 10
          <span class="navbar-breakpoint-down-d-none">items selected</span>
        </span>
      </li>
      <li class="nav-item">
        <button class="btn nav-btn btn-link">Select all</button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with descending sorting 1`] = `
<nav class="management-bar management-bar-light navbar navbar-expand-md">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-link nav-link-monospaced order-arrow-down-active btn-unstyled" aria-label="order-arrow">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-order-arrow">
            <title>order-arrow</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#order-arrow"></use>
          </svg>
        </button>
      </li>
    </ul>
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-btn nav-btn-monospaced btn-primary" aria-label="plus">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-plus">
            <title>plus</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#plus"></use>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with filters dropdown 1`] = `
<nav class="management-bar management-bar-light navbar navbar-expand-md">
  <div class="container">
    <ul class="navbar-nav">
      <li class="dropdown nav-item">
        <div class="dropdown">
          <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn nav-link navbar-breakpoint-down-d-none btn-unstyled">Filter and Order
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-bottom">
              <title>caret-bottom</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#caret-bottom"></use>
            </svg>
          </button>
          <form>
            <div class="dropdown-menu">
              <div class="dropdown-item">
                <div class="custom-control custom-checkbox custom-control-inline">
                  <label>
                    <input class="custom-control-input" name="item1checkbox" value="1" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description">Item 1</span>
                  </label>
                </div>
              </div>
              <div class="dropdown-item">
                <div class="custom-control custom-checkbox custom-control-inline">
                  <label>
                    <input class="custom-control-input" name="item2checkbox" value="2" ref="input" type="checkbox">
                    <span class="custom-control-indicator"></span>
                    <span class="custom-control-description">Item 2</span>
                  </label>
                </div>
              </div>
              <div class="dropdown-section">
                <button class="btn btn-block btn-primary" type="button">Done</button>
              </div>
            </div>
          </form>
        </div>
      </li>
      <li class="nav-item">
        <button class="btn nav-link nav-link-monospaced order-arrow-up-active btn-unstyled" aria-label="order-arrow">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-order-arrow">
            <title>order-arrow</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#order-arrow"></use>
          </svg>
        </button>
      </li>
    </ul>
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-btn nav-btn-monospaced btn-primary" aria-label="plus">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-plus">
            <title>plus</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#plus"></use>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with id 1`] = `
<nav class="management-bar management-bar-light navbar navbar-expand-md" id="myId">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-link nav-link-monospaced order-arrow-up-active btn-unstyled" aria-label="order-arrow">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-order-arrow">
            <title>order-arrow</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#order-arrow"></use>
          </svg>
        </button>
      </li>
    </ul>
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-btn nav-btn-monospaced btn-primary" aria-label="plus">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-plus">
            <title>plus</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#plus"></use>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with id in active state 1`] = `
<nav class="management-bar management-bar-primary navbar navbar-expand-md" id="myId">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <div class="custom-control custom-checkbox">
          <label>
            <input checked="" class="custom-control-input" ref="input" type="checkbox">
            <span class="custom-control-indicator"></span>
            <span class="custom-control-description sr-only">Select Page</span>
          </label>
        </div>
      </li>
      <li class="nav-item">
        <span class="navbar-text">1 of 10
          <span class="navbar-breakpoint-down-d-none">items selected</span>
        </span>
      </li>
      <li class="nav-item">
        <button class="btn nav-btn btn-link">Select all</button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with selection enabled 1`] = `
<nav class="management-bar management-bar-light navbar navbar-expand-md">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <div class="custom-control custom-checkbox">
          <label>
            <input class="custom-control-input" ref="input" type="checkbox">
            <span class="custom-control-indicator"></span>
            <span class="custom-control-description sr-only">Select Page</span>
          </label>
        </div>
      </li>
      <li class="nav-item">
        <button class="btn nav-link nav-link-monospaced order-arrow-up-active btn-unstyled" aria-label="order-arrow">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-order-arrow">
            <title>order-arrow</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#order-arrow"></use>
          </svg>
        </button>
      </li>
    </ul>
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-btn nav-btn-monospaced btn-primary" aria-label="plus">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-plus">
            <title>plus</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#plus"></use>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with state active 1`] = `
<nav class="management-bar management-bar-primary navbar navbar-expand-md">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <div class="custom-control custom-checkbox">
          <label>
            <input checked="" class="custom-control-input" ref="input" type="checkbox">
            <span class="custom-control-indicator"></span>
            <span class="custom-control-description sr-only">Select Page</span>
          </label>
        </div>
      </li>
      <li class="nav-item">
        <span class="navbar-text">1 of 10
          <span class="navbar-breakpoint-down-d-none">items selected</span>
        </span>
      </li>
      <li class="nav-item">
        <button class="btn nav-btn btn-link">Select all</button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render a management toolbar with view types 1`] = `
<nav class="management-bar management-bar-light navbar navbar-expand-md">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-link nav-link-monospaced order-arrow-up-active btn-unstyled" aria-label="order-arrow">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-order-arrow">
            <title>order-arrow</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#order-arrow"></use>
          </svg>
        </button>
      </li>
    </ul>
    <ul class="navbar-nav">
      <li class="dropdown nav-item">
        <div class="dropdown">
          <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn nav-link nav-link-monospaced btn-unstyled">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-list">
              <title>list</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#list"></use>
            </svg>
          </button>
          <div class="dropdown-menu dropdown-menu-indicator-start">
            <ul class="list-unstyled">
              <li>
                <a class=" dropdown-item active">
                  <div class="dropdown-item-indicator">
                    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-list">
                      <title>list</title>
                      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#list"></use>
                    </svg>
                  </div>List</a>
              </li>
              <li>
                <a class=" dropdown-item disabled" href="javascript:;">
                  <div class="dropdown-item-indicator">
                    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-table">
                      <title>table</title>
                      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#table"></use>
                    </svg>
                  </div>Table</a>
              </li>
              <li>
                <a class=" dropdown-item">
                  <div class="dropdown-item-indicator">
                    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-cards2">
                      <title>cards2</title>
                      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#cards2"></use>
                    </svg>
                  </div>Card</a>
              </li>
            </ul>
          </div>
        </div>
      </li>
      <li class="nav-item">
        <button class="btn nav-btn nav-btn-monospaced btn-primary" aria-label="plus">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-plus">
            <title>plus</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#plus"></use>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`ClayManagementToolbar should render the default markup 1`] = `
<nav class="management-bar management-bar-light navbar navbar-expand-md">
  <div class="container">
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-link nav-link-monospaced order-arrow-up-active btn-unstyled" aria-label="order-arrow">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-order-arrow">
            <title>order-arrow</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#order-arrow"></use>
          </svg>
        </button>
      </li>
    </ul>
    <ul class="navbar-nav">
      <li class="nav-item">
        <button class="btn nav-btn nav-btn-monospaced btn-primary" aria-label="plus">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-plus">
            <title>plus</title>
            <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#plus"></use>
          </svg>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;
